### DISCUSSION TITLE
Question: CLI available through maven repository?

### DISCUSSION THREAD

**Discussion by rdmueller** [structurizr/cli#106] [General] [answered]
The CLI is a good abstraction layer in front of the structurizr libs. But it seems that it is not available via maven.

It could be that I my thinking is wrong, but I currently think it would help to habe it avaialbe via maven.
For example, there is a [structurizr gradle plugin](https://github.com/jakzal/gradle-structurizr-cli) which downloads the jar first. This could be easier to implement via a dependency.

What do you think?

**Comment by simonbrowndotje** [comment, marked as answer]
I have no plans to release the CLI to Maven Central ... the CLI doesn't do much, so I'd recommend that any Gradle plugins (for example) just use the Structurizr libraries directly.

**Reply by rdmueller** [reply]
thanx for the quick answer. got the feeling from the source of the CLI that for instance an export is not just a handfull of lines of code. But I will recheck

**Reply by simonbrowndotje** [reply]
Here's a basic example of parsing a DSL and exporting to C4PlantUML.

```
    public static void main(String[] args) throws Exception {
        StructurizrDslParser parser = new StructurizrDslParser();
        parser.parse(new File("..."));
        Workspace workspace = parser.getWorkspace();
        ThemeUtils.loadThemes(workspace);
        C4PlantUMLExporter exporter = new C4PlantUMLExporter();
        Collection<Diagram> diagrams = exporter.export(workspace);
        for (Diagram diagram : diagrams) {
            System.out.println(diagram.getDefinition());
        }
    }
```

Dependencies:
 - structurizr-dsl
 - structurizr-export

**Reply by rdmueller** [reply]
oh, great. thanx a lot. Will give it a try! 🥳

**Reply by rdmueller** [reply]
it works - thanx!
will soon be part of docToolchain :-)
